ctutils/src/ct_option.rs

```diff
@@ -146,6 +195,13 @@ impl<T> CtOption<T> {
         optb
     }
 
+    /// Apply an additional [`Choice`] requirement to `is_some`.
+    #[inline]
+    pub const fn and_choice(mut self, is_some: Choice) -> Self {
+        self.is_some = self.is_some.and(is_some);
+        self
+    }
```

Comment:
I'm wondering if there's a better name we can use for this. It looks a lot like `CtOption::filter` but instead of having a predicate callback, it just takes an option.

Maybe `filter_as`?

Reply:
Went with `filter_by`